keras/src/utils/model_visualization_test.py

```diff
@@ -0,0 +1,44 @@
+from bs4 import BeautifulSoup
+
+import keras
+from keras.src import testing
+from keras.src.utils import model_to_dot
+
+
+class ModelVisualizationTest(testing.TestCase):
+    def _get_node_text(self, node):
+        attributes = node.get_attributes()
+
+        if "label" in attributes:
+            html = node.get_attributes()["label"]
+            parsed_html = BeautifulSoup(html)
```

Comment:
Rather than requiring an additional dependency, you should be able to extract text via a regex (since we only need to process a fixed, tiny subset of html, that is guaranteed to be constant over time).

Reply:
@fchollet 
Actually, the reason why I added pydot to requirement is not relevant to BeautifulSoup.
Without pydot library, I couldn't call "model_to_dot" method, since it uses this.
So, if I could use parsing library like "BeautifulSoup",because it is already  included,
I think it would be better to use this rather than using regex.